js/ui/floating_action_button/action_button_base.js

```diff
@@ -0,0 +1,228 @@
+        const actions = this.option("actions");
+        const minActionButtonCount = 2;
+        const lastActionIndex = actions.length - 1;
+
+        if(this._actionItems.length) {
+            this._actionItems.forEach(actionItem => {
+                actionItem.dispose();
+                actionItem.element().remove();
+            });
+        }
+
+        if(actions.length < minActionButtonCount) {
+            this._render();
+            return;
+        }
+
+        for(let i = 0; i < actions.length; i++) {
+            const action = actions[i];
+            const $actionElement = $("<div>")
+                .appendTo(getSwatchContainer(action.$element()));
+
+            eventsEngine.off($actionElement, "click");
+            eventsEngine.on($actionElement, "click", () => {
+                this._clickHandler();
+            });
+
+            const actionOffsetY = this.initialOption("indent") + this.initialOption("childIndent") * i;
+
+            action._options.position = {
+                of: this.$content(),
+                at: "center",
+                my: "center",
+                offset: {
+                    x: 0,
+                    y: -actionOffsetY
+                }
+            };
+
+            action._options.animation.show.delay = action._options.animation.show.duration * i;
+            action._options.animation.hide.delay = action._options.animation.hide.duration * (lastActionIndex - i);
+
+            this._actionItems.push(this._createComponent($actionElement, ActionButtonItem, action._options));
+        }
+    },
+
+    _optionChanged(args) {
+        switch(args.name) {
+            case "actions":
+                this._renderClick();
+                this._renderActions();
+                break;
+            case "maxActionButtonCount":
+                this._renderActions();
+                break;
+            case "closeIcon":
+                this._renderCloseIcon();
+                break;
+            case "position":
+                break;
+            default:
+                this.callBase(args);
+        }
+    }
+});
+
+exports.initAction = function(newAction) {
+    let isActionExist = false;
+    if(!actionButtonBase) {
+        const $fabMainElement = $("<div>")
+            .appendTo(getSwatchContainer(newAction.$element()));
+
+        actionButtonBase = new ActionButtonBase($fabMainElement,
+            extend({}, newAction._options, {
+                actions: [ newAction ],
+                visible: true
+            })
+        );
+
+    } else {
+        const savedActions = actionButtonBase.option("actions");
+
+        savedActions.forEach(action => {
+            if(action._options.id === newAction._options.id) {
+                isActionExist = true;
+                return newAction;
+            }
+        });
+
+        if(!isActionExist) {
+            if(savedActions.length >= actionButtonBase.option("maxActionButtonCount")) {
+                errors.log("W1014");
```

Comment:
A new button will not be rendered, but dxFloatingActionButton instance left initialized on the dom element. Should we dispose of it?

Reply:
Yes, done